frontend/src/components/auth/TeacherSignup/TeacherSignup.tsx

```diff
@@ -42,51 +53,125 @@ const renderPageComponent = (
       return <TeacherSignupThree {...teacherSignupProps} />;
     case 4:
       return <TeacherSignupFour {...teacherSignupProps} />;
+    case 5:
+      return <TeacherSignupFive />;
     default:
       return <></>;
   }
 };
+
 const TeacherSignup = (): React.ReactElement => {
-  const { authenticatedUser } = useContext(AuthContext);
-  const {
-    register,
-    handleSubmit,
-    setValue,
-    watch,
-    formState: { errors },
-  } = useForm<TeacherSignupForm>({ defaultValues, mode: "onChange" });
+  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+  const methods = useForm<TeacherSignupForm>({
+    defaultValues,
+    mode: "onChange",
+  });
   const [page, setPage] = React.useState(1);
+  const [user, setUser] = React.useState<AuthenticatedUser>();
+
+  const onSuccessfulSignup = () => {
+    if (user) {
+      setAuthenticatedUser(user);
+      setPage(5);
+    }
+  };
+
+  const [registerTeacher] = useMutation<{ register: AuthenticatedUser }>(
+    REGISTER_TEACHER,
+  );
+  const [addTeacherToSchool] = useMutation<{
+    addTeacherToSchool: SchoolResponse;
+  }>(ADD_TEACHER_TO_SCHOOL, {
+    onCompleted() {
+      onSuccessfulSignup();
+    },
+  });
+  const [getSchool] = useLazyQuery<{ school: SchoolResponse }>(GET_SCHOOL, {
+    onCompleted: async (data) => {
+      await addTeacherToSchool({
```

Comment:
i'm a bit confused by this! every time we query for a school, do we also call `addTeacherToSchool`?